src/platform/posix.c

```diff
@@ -876,4 +877,14 @@ datetime64 platform_get_datetime_now_utc()
 	return utcNow;
 }
 
+utf8* platform_get_username() {
+	struct passwd* pw = getpwuid(getuid());
+
+	if (pw) {
+		return pw->pw_name;
+	} else {
+		return language_get_string(STR_PLAYER_DEFAULT_NAME);
```

Comment:
I think this should return NULL. ```STR_PLAYER_DEFAULT_NAME``` is specific to multiplayer not getting the current user name from the operating system.